client/src/App.js

```diff
@@ -35,73 +35,97 @@ library.add(
 class App extends Component {
   state = {
     userInfo: {},
-    isLogin: true
+    isLogin: true,
   };
   setUserInfo = userInfo => {
     this.setState({ userInfo, isLogin: true });
   };
+
+  clearUserInfo = () => {
+    this.setState({ userInfo: '', isLogin: false });
+  }
+
   render() {
     const { isLogin } = this.state;
     return (
       <BrowserRouter>
         {isLogin ? (
```

Comment:
@IsraaSulaiman I have an idea for the future, or for the next app - you might be able to combine your two `<Switch>` statements into one, and put all of the routes inside them.  Then you wouldn't have to have a large if-statement inside render() here, which might make this component a lot simpler.
To do that, you would probably need to create a "LoggedOutRoute" component to use for /login, which redirects if the user is already logged in (so it behaves in an opposite way to PrivateRoute), and also clearUserInfo() might need to call history.push('/login') so that when the user logs out, they are redirected to the login page.
No need to do all of this now, but it's an alternate way to create your react app routes in the future which you might like 🤷‍♂️

Reply:
@IsraaSulaiman  we have been looking for something like this :smile:

Thanks, @tomduggan85 :pray: